addon/components/base/bs-form/element.js

```diff
@@ -617,11 +639,16 @@ export default FormGroup.extend({
    * @type string
    * @private
    */
-  validation: computed('hasCustomError', 'hasErrors', 'hasWarnings', 'hasValidator', 'showValidation', 'isValidating', 'disabled', function() {
+  validation: computed('hasCustomError', 'hasErrors', 'hasCustomWarning', 'hasWarnings', 'hasValidator', 'showValidation', 'isValidating', 'disabled', function() {
     if (!this.get('showValidation') || !this.get('hasValidator') || this.get('isValidating') || this.get('disabled')) {
       return null;
+    } else if (this.get('showOwnValidation') || this.get('showAllValidations')) {
+      // The display of validation messages has been enabled
+      return this.get('hasErrors') || this.get('hasCustomError') ? 'error' : (this.get('hasWarnings') || this.get('hasCustomWarning') ? 'warning' : 'success');
+    } else {
+      // If there are custom errors or warnings these should always be shown
+      return this.get('hasCustomError') ? 'error' : (this.get('hasWarnings') || this.get('hasCustomWarning') ? 'warning' : null);
```

Comment:
Related to the previous comment, the `null` return value here can never be reached, I think? If this statement is reached, either `hasCustomError` or `hasCustomWarning` has to be true.